### PR TITLE
Demos: process new amd demos

### PR DESCRIPTION
**Pull request by arschmitz**
This accompanies jquery/jquery-ui#1557 to process the new demo infrastructure

### CODE REVIEW

**Comment by jzaefferer** [comment]
Fixes #112

**Comment by arschmitz** [comment]
Updated to account for there no longer being a `require-config` tag

**Comment by jzaefferer** [comment]
Diff looks good, still haven't tested anything locally. I don't know how to configure DB to use your `amd-demos` branch from your repo.

File: Gruntfile.js [hunk 1]
```diff
@@ -157,26 +161,39 @@ grunt.registerTask( "build-demos", function() {
...
+				stable.pkg.version + "/jquery-ui.js\"></script>" );
+
+		// Any modules that are not part of the actual library must contain .js and are added as
+		// new script tags before the bootstrap file
+		if ( i18n ) {
+			i18n = i18n.match( /\S+/g ).filter( function( value ) {
+				return /i18n/.test( value );
+			} );
+			if ( i18n.length ) {
+				for( var i = 0; i < i18n.length; i++ ) {
```
Comment (scottgonzalez) [inline Gruntfile.js]: Missing space after `for`.

Comment (scottgonzalez) [inline Gruntfile.js]: Actually, just switch to `i18n.forEach()` and drop the conditional for length.

Comment (arschmitz) [inline Gruntfile.js]: Oh yeah i can actually use `forEach` this is node :-)

File: Gruntfile.js [hunk 2]
```diff
@@ -157,26 +161,39 @@ grunt.registerTask( "build-demos", function() {
...
-		source = source.replace(
-			/<script src="\.\.\/\.\.\/ui\/[^>]+>/,
-			"<script src=\"//code.jquery.com/ui/" + stable.pkg.version + "/jquery-ui.js\">" );
-		source = source.replace(
-			/^.*<script src="\.\.\/\.\.\/ui\/[^>]+><\/script>\n/gm,
-			"" );
+		bootstrap
+			.removeAttr( "src" )
+			.html( "\n\t$( function() {" + bootstrap.html() + "} );\n\t" )
+			.removeAttr( "data-modules" );
```
Comment (scottgonzalez) [inline Gruntfile.js]: Rather than modifying the two existing scripts, I'd rather generate new elements and remove the existing ones. It's less fragile.

Comment (arschmitz) [inline Gruntfile.js]: That makes the output ugly because cheerio will keep the whitespace.

Comment (arschmitz) [inline Gruntfile.js]: Actually i can use `replaceWith`.

Comment (scottgonzalez) [inline Gruntfile.js]: oh, ok

**Comment by jzaefferer** [comment]
Was there a reason for keeping this open? If not, can you land this in a 1-12 branch?

**Comment by jzaefferer** [comment]
Alex confirmed that this can land, in a 1-12 branch.

**Comment by jzaefferer** [comment]
Merge to 1-12 branch, see 2d3e7c05a888aa4df4f69afc7cc1eb502c43bc74

**Pull request closed without merging**